From 29433495d697e4dcb3bc50ff0e0d866acb949890 Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Wed, 11 Jan 2017 17:53:32 -0800
Subject: [PATCH] qtdemux: free seqh after calling
 qtdemux_parse_svq3_stsd_data()

The seqh buffer allocated in qtdemux_parse_svq3_stsd_data() needs to
be freed by the caller after use.

https://bugzilla.gnome.org/show_bug.cgi?id=777157

Upstream-Status: Backport

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 gst/isomp4/qtdemux.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index e105809..7a431e0 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -10375,6 +10375,7 @@ qtdemux_parse_trak (GstQTDemux * qtdemux, GNode * trak)
              * than its own fourcc */
             gst_caps_set_simple (stream->caps, "seqh", GST_TYPE_BUFFER, seqh,
                 NULL);
+            gst_buffer_unref (seqh);
           }
 
           GST_DEBUG_OBJECT (qtdemux, "found codec_data in stsd");
-- 
1.9.1

